i18n/.js/src/main/scala/io/udash/i18n/bindings/TranslationBinding.scala

```diff
@@ -5,29 +5,42 @@ import io.udash.bindings.Bindings
 import io.udash.i18n.Translated
 import io.udash.logging.CrossLogging
 import org.scalajs.dom._
+import org.scalajs.dom.raw.Node
 import scalatags.JsDom.Modifier
 
 import scala.concurrent.Future
 import scala.util.{Failure, Success}
 
-private[i18n] class TranslationBinding(translation: Future[Translated], placeholder: Option[Element])
-  extends Modifier with Bindings with CrossLogging {
+private[i18n] class TranslationBinding(
+  translation: Future[Translated],
+  placeholder: Option[Element]
+) extends Modifier with Bindings with CrossLogging with TranslationBindingHelper {
 
-  override def applyTo(t: Element): Unit =
-    if (translation.isCompleted && translation.value.get.isSuccess) {
-      t.appendChild(document.createTextNode(translation.value.get.get.string))
-    } else {
-      val holder: Node = placeholder.getOrElse(emptyStringNode())
-      t.appendChild(holder)
+  protected def impl[T](placeholder: Option[T], append: T => T, replace: (T, T) => T)(
+    implicit str2T: String => T, node2T: Node => T
+  ): Unit = translation.value match {
+    case Some(Success(Translated(text))) =>
+      append(text)
 
+    case _ =>
+      val holder: T = placeholder.getOrElse(emptyStringNode())
+      append(holder)
       translation.onCompleteNow {
-        case Success(text) =>
-          t.replaceChild(
-            document.createTextNode(text.string),
-            holder
-          )
+        case Success(Translated(text)) =>
+          replace(text, holder)
         case Failure(ex) =>
           logger.error(ex.getMessage)
       }
-    }
+  }
+
+  override def applyTo(t: Element): Unit =
+    impl(placeholder, t.appendChild, t.replaceChild)
+}
+
+private[i18n] class RawTranslationBinding(
+  translation: Future[Translated],
+  placeholder: Option[Element]
+) extends TranslationBinding(translation, placeholder) {
+  override def applyTo(t: Element): Unit =
+    impl(placeholder.map(Seq(_)), t.appendKids, t.replaceKids)
```

Comment:
You might want to use `io.udash.bindings.modifiers.ElementExts#replaceChildren` for multiple nodes replacement and inline appending. It should work in both cases if we assume `impl` works on `Seq[Node]`.

Reply:
Good idea, thanks!